### ISSUE TITLE
Suggestion to add total water consumption counter

### ISSUE THREAD

**Issue by chrjoedal** [weissm/homeassistant-grohe_sense#2]
Hi,

Thank you for continuing the work on this project!
May I suggest to make a total water consumption counter, with the correct device type etc., so it is possible to add the sensor to the energy dashboard?
See: https://www.home-assistant.io/docs/energy/faq/#troubleshooting-missing-entities

I guess you have all the data available, as you have the daily and weekly consumption, but it needs to be a total counter if I read the documentation correctly

Thank you!
Best,
Christian Schou Jødal

**Comment by xxpas02** [comment]
Yes, I don’t know what’s required to achieve it, but it would be very useful to have a device/entity that meets the requirements for a Water Source in the Energy Dashboard.

**Comment by dreimer1986** [comment]
I have zero clue if the sensor or rather the Grohe API is doing a job good enough for a sensor for the Energy Dashboard, but I added it anyways. I already did such trickery way too often with other sensors to follow spcific needs. My first one for example was modifying the supported feature levels of all my covers to work fine with voice assistants and now I did the same basic idea here.

Your solution is the underrated customize.yaml file found here: /homeassistant/customize.yaml where you add this block to make HA realize that there is a nice sensor for the energy panel:

```
sensor.mein_guard_flowrate:
  device_class: water
  state_class: total
```

**Comment by vincedarley** [comment]
Isn't there one already? I use "sensor.water_guard_water_consumption" in my Energy dashboard and it works just fine...

**Comment by dreimer1986** [comment]
I don't have this one. Where did you get it from?

<img width="1436" height="633" alt="Image" src="https://github.com/user-attachments/assets/5ec1cfee-e6d5-4f7c-a91f-1575be53b3f5" />

**Comment by vincedarley** [comment]
<img width="1029" height="928" alt="Image" src="https://github.com/user-attachments/assets/1cfb60d6-a3c0-4b20-8368-6b2a81584b4a" />

I'm possibly on a different fork: https://github.com/Flo-Schilli/ha-grohe_smarthome
